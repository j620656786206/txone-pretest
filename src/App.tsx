import {
  TonicProvider,
  PortalManager, // allows you to create and manage portals in the application
  // ToastManager, // allows you to create and manage toasts in the application
} from '@tonic-ui/react'
import Layout from './layouts/Layout'
import MenuIcon from './components/MenuIcon'

import './App.css'
import styles from './styles.module.scss'

function App() {
  return (
    <TonicProvider
      colorMode={{
        defaultValue: 'dark', // One of: 'dark', 'light'
      }}
      useCSSBaseline={true} // If `true`, apply CSS reset and base styles
    >
      <PortalManager>
        <Layout>
          <div className={styles.body}>
            <MenuIcon />
          </div>
        </Layout>
      </PortalManager>
    </TonicProvider>
  )
}

export default App
