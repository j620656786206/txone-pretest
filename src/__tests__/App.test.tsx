import { render, screen, act } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import App from '../App'

describe('App', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<App />)

    expect(baseElement).toBeTruthy()
  })

  it('should match snapshot', () => {
    const { baseElement} = render(<App />)

    expect(baseElement).toMatchSnapshot()
  })

  it('click button, should match snapshot', async () => {
    const user = userEvent.setup()
    const { baseElement } = render(<App />)

    // eslint-disable-next-line testing-library/no-unnecessary-act
    await act(async () => {
      const button = screen.getByRole('button')
      await user.click(button)
    })

    expect(baseElement).toMatchSnapshot()
  })

  it('click button, list items shoud be truthy', async () => {
    const user = userEvent.setup()
    render(<App />)

    // eslint-disable-next-line testing-library/no-unnecessary-act
    await act(async () => {
      const button = screen.getByRole('button')
      await user.click(button)
    })
    const settingsItem = screen.getByText('Settings')
    const accountsItem = screen.getByText('Accounts')
    const privacyControlItem = screen.getByText('Privacy control')

    expect(settingsItem).toBeTruthy()
    expect(accountsItem).toBeTruthy()
    expect(privacyControlItem).toBeTruthy()
  })
})
