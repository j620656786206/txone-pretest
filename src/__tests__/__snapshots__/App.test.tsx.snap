// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App click button, should match snapshot 1`] = `
<body>
  <div>
    <div
      class="body"
    >
      <div
        class="css-1y2pe6x-Box ebasi3p0"
      >
        <button
          aria-controls="@tonic-ui/react:Menu-4"
          aria-expanded="true"
          aria-haspopup="menu"
          class="css-xz8qfw-Box ebasi3p0"
          id="@tonic-ui/react:MenuToggle-4"
          role="button"
          tabindex="0"
          type="button"
        >
          <div
            class="css-wj810m-Box ebasi3p0"
          >
            <svg
              aria-hidden="true"
              class="css-1021at6-Box-SVGIconBase ei4t6id0"
              focusable="false"
              role="presentation"
              viewBox="0 0 16 16"
            >
              <g>
                <path
                  d="M14.104 13h-1.231c-0.381-1.055-0.958-1.956-1.693-2.702l0.001 0.001c-0.637-0.638-1.378-1.174-2.195-1.578l-0.048-0.022c0.658-0.708 1.062-1.661 1.062-2.707 0-1.655-1.010-3.075-2.447-3.676l-0.026-0.010c0.547-0.795 1.452-1.31 2.477-1.31 1.655 0 2.996 1.342 2.996 2.996 0 1.189-0.693 2.216-1.696 2.701l-0.018 0.008v0.458c1.254 0.339 2.329 0.98 3.182 1.842l0.001 0.001c0.983 1 3.069 4-0.364 4zM7.286 9.158v-0.458c1.022-0.493 1.714-1.521 1.714-2.711 0-1.657-1.343-3-3-3s-3 1.343-3 3c0 1.19 0.692 2.218 1.696 2.703l0.018 0.008v0.458c-1.254 0.339-2.329 0.98-3.182 1.842l-0.001 0.001c-0.983 1-3.069 4 0.364 4h8.208c3.433 0 1.347-3 0.364-4-0.854-0.862-1.929-1.504-3.133-1.831l-0.049-0.011z"
                />
              </g>
            </svg>
          </div>
        </button>
        <div
          aria-labelledby="@tonic-ui/react:MenuToggle-4"
          class="css-17bdax9-Box ebasi3p0"
          data-menu-id="@tonic-ui/react:Menu-4"
          data-popper-escaped=""
          data-popper-placement="bottom-start"
          data-popper-reference-hidden=""
          id="@tonic-ui/react:Menu-4"
          role="menu"
          style="position: absolute; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 0px);"
          tabindex="-1"
        >
          <div
            class="css-g7qg7g-Box ebasi3p0"
          >
            <div
              class="css-1efsrtw-Box ebasi3p0"
            >
              <button
                class="css-1q4qc6j-Box ebasi3p0"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="css-i07n4o-Box-SVGIconBase ei4t6id0"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 16 16"
                >
                  <g>
                    <path
                      d="M16 9v-2l-2.13-0.27c-0.163-0.757-0.447-1.427-0.834-2.023l0.014 0.023 1.33-1.7-1.38-1.39-1.74 1.36c-0.573-0.376-1.242-0.664-1.96-0.823l-0.040-0.007-0.26-2.17h-2l-0.26 2.13c-0.757 0.163-1.427 0.447-2.023 0.834l0.023-0.014-1.67-1.31-1.41 1.42 1.34 1.69c-0.376 0.573-0.664 1.242-0.823 1.96l-0.007 0.040-2.17 0.25v2l2.14 0.27c0.163 0.757 0.447 1.427 0.834 2.023l-0.014-0.023-1.32 1.7 1.41 1.41 1.71-1.38c0.574 0.37 1.243 0.65 1.961 0.803l0.039 0.007 0.24 2.19h2l0.27-2.14c0.757-0.163 1.427-0.447 2.023-0.834l-0.023 0.014 1.73 1.34 1.38-1.38-1.38-1.75c0.373-0.573 0.657-1.243 0.813-1.96l0.007-0.040zM8 11c-1.657 0-3-1.343-3-3s1.343-3 3-3c1.657 0 3 1.343 3 3v0c0 1.657-1.343 3-3 3v0z"
                    />
                  </g>
                </svg>
                <div
                  class="css-1efsrtw-Box ebasi3p0"
                >
                  Settings
                </div>
              </button>
              <button
                class="css-1q4qc6j-Box ebasi3p0"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="css-i07n4o-Box-SVGIconBase ei4t6id0"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 16 16"
                >
                  <g>
                    <path
                      d="M14.104 13h-1.231c-0.381-1.055-0.958-1.956-1.693-2.702l0.001 0.001c-0.637-0.638-1.378-1.174-2.195-1.578l-0.048-0.022c0.658-0.708 1.062-1.661 1.062-2.707 0-1.655-1.010-3.075-2.447-3.676l-0.026-0.010c0.547-0.795 1.452-1.31 2.477-1.31 1.655 0 2.996 1.342 2.996 2.996 0 1.189-0.693 2.216-1.696 2.701l-0.018 0.008v0.458c1.254 0.339 2.329 0.98 3.182 1.842l0.001 0.001c0.983 1 3.069 4-0.364 4zM7.286 9.158v-0.458c1.022-0.493 1.714-1.521 1.714-2.711 0-1.657-1.343-3-3-3s-3 1.343-3 3c0 1.19 0.692 2.218 1.696 2.703l0.018 0.008v0.458c-1.254 0.339-2.329 0.98-3.182 1.842l-0.001 0.001c-0.983 1-3.069 4 0.364 4h8.208c3.433 0 1.347-3 0.364-4-0.854-0.862-1.929-1.504-3.133-1.831l-0.049-0.011z"
                    />
                  </g>
                </svg>
                <div
                  class="css-1efsrtw-Box ebasi3p0"
                >
                  Accounts
                </div>
              </button>
              <button
                class="css-1q4qc6j-Box ebasi3p0"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="css-i07n4o-Box-SVGIconBase ei4t6id0"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 16 16"
                >
                  <g>
                    <path
                      d="M13.25 6h-1.25v-3c-0.002-1.656-1.344-2.998-3-3h-2c-1.656 0.002-2.998 1.344-3 3v3h-1.25c-0.414 0.002-0.748 0.336-0.75 0.75v8.5c0.002 0.414 0.336 0.748 0.75 0.75h10.5c0.414-0.002 0.748-0.336 0.75-0.75v-8.5c-0.002-0.414-0.336-0.748-0.75-0.75h-0zM9 12h-2v-2h2zM10 6h-4v-3c0.001-0.552 0.448-0.999 1-1h2c0.552 0.001 0.999 0.448 1 1v0z"
                    />
                  </g>
                </svg>
                <div
                  class="css-1efsrtw-Box ebasi3p0"
                >
                  Privacy control
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`App should match snapshot 1`] = `
<body>
  <div>
    <div
      class="body"
    >
      <div
        class="css-1y2pe6x-Box ebasi3p0"
      >
        <button
          aria-controls="@tonic-ui/react:Menu-3"
          aria-haspopup="menu"
          class="css-xz8qfw-Box ebasi3p0"
          id="@tonic-ui/react:MenuToggle-3"
          role="button"
          tabindex="0"
          type="button"
        >
          <div
            class="css-wj810m-Box ebasi3p0"
          >
            <svg
              aria-hidden="true"
              class="css-1021at6-Box-SVGIconBase ei4t6id0"
              focusable="false"
              role="presentation"
              viewBox="0 0 16 16"
            >
              <g>
                <path
                  d="M14.104 13h-1.231c-0.381-1.055-0.958-1.956-1.693-2.702l0.001 0.001c-0.637-0.638-1.378-1.174-2.195-1.578l-0.048-0.022c0.658-0.708 1.062-1.661 1.062-2.707 0-1.655-1.010-3.075-2.447-3.676l-0.026-0.010c0.547-0.795 1.452-1.31 2.477-1.31 1.655 0 2.996 1.342 2.996 2.996 0 1.189-0.693 2.216-1.696 2.701l-0.018 0.008v0.458c1.254 0.339 2.329 0.98 3.182 1.842l0.001 0.001c0.983 1 3.069 4-0.364 4zM7.286 9.158v-0.458c1.022-0.493 1.714-1.521 1.714-2.711 0-1.657-1.343-3-3-3s-3 1.343-3 3c0 1.19 0.692 2.218 1.696 2.703l0.018 0.008v0.458c-1.254 0.339-2.329 0.98-3.182 1.842l-0.001 0.001c-0.983 1-3.069 4 0.364 4h8.208c3.433 0 1.347-3 0.364-4-0.854-0.862-1.929-1.504-3.133-1.831l-0.049-0.011z"
                />
              </g>
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</body>
`;
